misc/helper-scripts/rosmsg_to_asn1.py

```diff
@@ -0,0 +1,417 @@
+            if asn1_type not in import_libraries[lib_asn]:
+                import_libraries[lib_asn].append(asn1_type)
+        else:
+            import_libraries[lib_asn] = [asn1_type]
+        slot_asn1_types.append(asn1_type)
+        field_types.append(field_type)
+        lengths.append(length)
+        if not primitive:
+            msg_deps.append(slot_type)
+
+    msg_string += generate_libraries_string(import_libraries)
+    fixed_indices = [i for i, x in enumerate(field_types)
+                     if x == "fixed-length"]
+    variable_indices = [i for i, x in enumerate(field_types)
+                        if x == "variable-length"]
+
+    for i in fixed_indices:
+        msg_string += "L" + slots[i].replace("_", "-") + "::="
+        msg_string += " SEQUENCE (SIZE(0.." + str(lengths[i])+")) OF "
+        msg_string += slot_asn1_types[i] + "\n"
+        field_types[i] = "simple"
+        slot_asn1_types[i] = "L" + slots[i].replace("_", "-")
+
+    for i in variable_indices:
+        msg_string += "V" + slots[i].replace("_", "-") + "::="
+        msg_string += " SEQUENCE (SIZE(0..256)) OF "
+        msg_string += slot_asn1_types[i] + "\n"
+        field_types[i] = "simple"
+        slot_asn1_types[i] = "V" + slots[i].replace("_", "-")
+
+    if (num_fields == 1):
+        if field_types[0] == "simple":
+            if slots[0] == "type":
+                slots[0] += "-" + slot_asn1_types[0]
+            if slots[0].lower() == slot_asn1_types[0].lower():
+                slots[0] += "-field"
+            msg_string += msg_name + "::=\nSEQUENCE\n{\n"
+            msg_string += "\t" + slots[0].replace("_", "-") + "\t"
+            msg_string += slot_asn1_types[0] + "\n}\n"
+    else:
+        msg_string += msg_name + "::=\nSEQUENCE\n{\n"
+        comma = ""
+        for i in range(num_fields):
+            if field_types[i] == "simple":
+                if slots[i] == "type":
+                    slots[i] += "-" + slot_asn1_types[i]
+                if slots[i].lower() == slot_asn1_types[i].lower():
+                    slots[i] += "-field"
+                msg_string += comma + "\t" + slots[i].replace("_", "-")
+                msg_string += "\t" + slot_asn1_types[i]
+                comma = ",\n"
+        msg_string += "\n}\n"
+    msg_string += "END"
+
+    return msg_string, msg_deps
+
+
+def save_msg(asn1_str, msg, out_dir):
+    """
+    Save the ASN.1 generated message to a file.
+    :param asn1_str: The message string.
+    :param msg: The message name.
+    :param out_dir: The output directory.
+    """
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    if "/" in msg:
+        msg = msg.split("/")[1]
+    f = open(out_dir + msg + ".asn", "w")
+    f.write(asn1_str)
+    f.close()
+
+
+def main():
+    # Get the messages to be generated
+    args = parse_args()
+    print(args)
+    messages = args.messages
+    out_dir = args.output
+    if out_dir[-1] != "/":
+        out_dir += "/"
```

Comment:
Do you want to accept any out_dir specified, or would it make sense to sanitize this? eg. what with `out_dir == ''`?

Reply:
Shouldn't matter as it is not intended to be used standalone.  Added something for the empty case nevertheless.

Reply:
:+1: